app/controllers/admin/staff_action_logs_controller.rb

```diff
@@ -34,26 +34,18 @@ def diff
     prev = JSON.parse(prev) if prev
     cur = JSON.parse(cur) if cur
 
-    diff_fields = {}
-
     output = +"<h2>#{CGI.escapeHTML(cur&.dig("name").to_s)}</h2><p></p>"
 
+    diff_fields = {}
     diff_fields["name"] = { prev: prev&.dig("name").to_s, cur: cur&.dig("name").to_s }
 
-    %w[default user_selectable].each do |f|
-      diff_fields[f] = { prev: (!!prev&.dig(f)).to_s, cur: (!!cur&.dig(f)).to_s }
+    case @history.action
+    when *%i[change_theme delete_theme].map { |i| UserHistory.actions[i] }
+      diff_theme(diff_fields, prev, cur)
+    when *%i[tag_group_create tag_group_destroy tag_group_change].map { |i| UserHistory.actions[i] }
+      diff_tag_group(diff_fields, prev, cur)
```

Comment:
I guess we don't need it, because these two cases have covered all situations, and other situations cannot be handled (other `UserHistory` is likely not stored in JSON format, which will cause `ParseError`)